### PR TITLE
feat: refurbish bottles

### PR DESCRIPTION
**Pull request by bahrmichael**
This PR introduces a mechanic to return dirty bottles that players can refurbish into new bottles.

Bottles are returned to Bottle Return Stations with a mechanism similar to the city debt for treasuries.

Players can then use the recipe "clean bottles" or "clean bottles with soap" to get regular bottles again.

Todos:

- ~[ ] Notify the player via a one-time message that dirty bottles are returned in the markets. This is a new mechanic and may clog up the markets.~
- [x] Introduce bottle return entity where citizens return dirty bottles.
- ~[ ] Consider refurbishment plants instead of letting players refurbish bottles in assembling machines.~ I felt like mandatory plants are not a good fit yet, and would rather let the player build the recycling setups themselves. Will reconsider larger plants for boosted production, e.g. steel mills.

### CODE REVIEW

File: util.lua [hunk 1]
```diff
@@ -104,6 +104,29 @@ local function findCityByEntityUnitNumber(unitNumber)
     return cityName or "Unknown"
 end
 
+
+local function list_special_city_buildings(city, name)
+    local entities = {}
+    if city.special_buildings.other[name] ~= nil and #city.special_buildings.other[name] > 0 then
+        entities = city.special_buildings.other[name]
+    else
+        entities = game.surfaces[Constants.STARTING_SURFACE_ID].find_entities_filtered{
```
Comment (winex) [inline util.lua:113]: ~~Constants.STARTING_SURFACE_ID~~ `city.surface_index`

**Comment by winex** [comment]
i think returning to markets is not a good idea -- that will block them 100%.
didn't you think about some new garbage-piling entity?

File: util.lua [hunk 2]
```diff
@@ -209,6 +209,28 @@ local function findCityByEntityUnitNumber(unitNumber)
...
 end
 
+local function list_special_city_buildings(city, name)
+    local entities = {}
+    if city.special_buildings.other[name] ~= nil and #city.special_buildings.other[name] > 0 then
+        entities = city.special_buildings.other[name]
+    else
+        entities = game.surfaces[city.surface_index].find_entities_filtered{
+            name=name,
+            position=city.special_buildings.town_hall.position,
```
Comment (winex) [inline util.lua:219]: oh, how this `LuaEntity` access got into here, again?! this:
```diff
-position=city.special_buildings.town_hall.position,
+position=city.center,
```

**Comment by bahrmichael** [comment]
> i think returning to markets is not a good idea -- that will block them 100%. yeah, mixing input/output is always wrong.
didn't you think about some new garbage-piling entity?

Thought about it, and am leaning towards a dedicated entity now. Had a similar problem with train stations, where I introduced GUI settings to filter passengers. Here it's more reasonable to introduce a new entity.

File: util.lua (hunk 2 above)
Comment (bahrmichael) [reply util.lua:219]: That branch is a couple weeks old, so there may be some old stuff.

**Pull request merged**